### PR TITLE
fix(36989): 'async' modifier cannot be used in an ambient context.ts

### LINKED ISSUE #36989: 'async' modifier cannot be used in an ambient context.ts
**TypeScript Version:**  3.8.2

**Search Terms:** allowJs declaration  'async' modifier cannot be used in an ambient context.ts

**Code**
see [code](https://github.com/francesc79/test-ts)

when run tsc with allowJs and declaration flags enabled the code below
```TestJS.js
export class TestJS {
    callAsync() {
        return Promise.resolve(1);
    }

    async test() {
        await this.callAsync();
    }
}
```
produce the declaretion:
```TestJS.d.ts
export class TestJS {
    callAsync(): Promise<number>;
    async test(): Promise<void>; --> error
}
```
but when a try to use this class I receive 'async' modifier cannot be used in an ambient context.ts(1040)

**Expected behavior:**
the TestJS.d.ts shouldn't contain async 

**Actual behavior:**
TestJS.d.ts conteins async 

**Playground Link:** [code](https://github.com/francesc79/test-ts)

**Related Issues:**

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #36989

### CODE REVIEW

**Comment by boneskull** [comment]
Was hoping somebody could trigger a build of this.

**Comment by RyanCavanaugh** [comment]
@rbuckton can you review? Looking at taking this into 3.8.3

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -6204,7 +6204,7 @@ namespace ts {
                             && isTypeIdenticalTo(getTypeOfSymbol(p), getTypeOfPropertyOfType(baseType, p.escapedName)!))) {
                             return [];
                         }
-                        const flag = modifierFlags | (isStatic ? ModifierFlags.Static : 0);
+                        const flag = (modifierFlags & ModifierFlags.Async ? modifierFlags & ~ModifierFlags.Async : modifierFlags) | (isStatic ? ModifierFlags.Static : 0);
```
Comment (weswigham) [inline src/compiler/checker.ts]:
```suggestion
                        const flag = (modifierFlags & ~ModifierFlags.Async) | (isStatic ? ModifierFlags.Static : 0);

```

you don't need to bother with the check~

Comment (a-tarasyuk) [reply src/compiler/checker.ts]: @weswigham Oke. Thanks!

**Comment by weswigham** [comment]
@typescript-bot cherry-pick this into release-3.8

**Pull request merged**